golem-base/storageutil/entity/store_entity.go

```diff
@@ -12,6 +13,12 @@ import (
 	"github.com/ethereum/go-ethereum/golem-base/storageutil/keyset"
 )
 
+// This regex should not allow $ as the first character, since we use that for
+// special meta-annotations like $owner.
+const AnnotationIdentRegex string = `[a-zA-Z_][a-zA-Z0-9_]*`
+
+var annotationIdentRegexCompiled *regexp.Regexp = regexp.MustCompile(fmt.Sprintf("^%s$", AnnotationIdentRegex))
```

Comment:
why not putting `^[a-zA-Z_][a-zA-Z0-9_]*$` into  `AnnotationIdentRegex` to begin with?